charts/terra-argocd-app/templates/app.yaml

```diff
@@ -1,39 +1,45 @@
-{{- with .Values -}}
+{{- if .Values.releaseType | eq "app" }}
+# Generates an ArgoCD application for a Terra application release
+{{- $app := required "releaseName is required" .Values.releaseName }}
+{{- $env := required "targetName is required" .Values.targetName }}
 apiVersion: argoproj.io/v1alpha1
 kind: Application
 metadata:
-  name: {{ required "A valid app name is required" .app }}-{{ required "A valid environment is required" .environment }}
+  name: "{{ required "A valid releaseName is required" .Values.releaseName }}-{{ required "A valid targetName is required" .Values.targetName }}"
```

Comment:
I'm a little confused here. On line 3, we're setting $app to .Values.releaseName. Can we not refer to $app here? I think it'll help with the readability and uniformity.

Reply:
Adjusted!